autoscheduler/scheduler/tests/api_tests.py

```diff
@@ -0,0 +1,132 @@
+from unittest.mock import patch
+from datetime import time
+from rest_framework.test import APITestCase, APIClient
+from scheduler.views import (_parse_course_filter, _parse_unavailable_times,
+                             _serialize_schedule)
+from scheduler.utils import UnavailableTime, CourseFilter
+from scraper.models import Section, Instructor
+from scraper.serializers import SectionSerializer
+
+class SchedulingAPITests(APITestCase):
+    """ Tests for the functionality in scheduling.views """
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.client = APIClient()
+
+        instructor = Instructor(id="name")
+        instructor.save()
+        cls.sections = [
+            Section(id=1, crn=1, subject='CSCE', course_num='121', section_num='501',
+                    term_code='201931', min_credits=0, honors=False, web=False,
+                    current_enrollment=0, max_enrollment=0, instructor=instructor),
+            Section(id=2, crn=2, subject='CSCE', course_num='221', section_num='501',
+                    term_code='201931', min_credits=0, honors=False, web=False,
+                    current_enrollment=0, max_enrollment=0, instructor=instructor),
+        ]
+        Section.objects.bulk_create(cls.sections)
+
+    def test_parse_course_filter_is_correct(self):
+        """ Tests that _parse_counter_filter works on a typical input """
+
+        # Arrange
+        courses = [
+            {
+                "subject": "CSCE",
+                "courseNum": "121",
+                "sections": ["500"],
+                "honors": False,
+                "web": False,
+            },
+        ]
+
+        expected = [
+            CourseFilter(subject="CSCE", course_num="121", section_nums=["500"],
+                         honors=False, web=False)
+        ]
+
+        # Act
+        result = _parse_course_filter(courses)
+
+        # Assert
+        self.assertEqual(result, expected)
+
+    def test_parse_unavailable_times_is_correct(self):
+        """ Tests that _parse_unavailable_times works on a typical input """
+
+        # Arrange
+        availabilities = [
+            {
+                "startTime": "0800",
+                "endTime": "1000",
+                "day": 0,
+            },
+        ]
+
+        expected = [
+            UnavailableTime(time(8, 00), time(10, 00), 0)
+        ]
+
+        # Act
+        result = _parse_unavailable_times(availabilities)
+
+        # Assert
+        self.assertEqual(result, expected)
+
+    def test_serialize_schedule_is_correct(self):
+        """ Tests that _serialize_schedule works on a typical input """
+
+        # Arrange
+        schedule = [("1", "2")]
+
+        expected = [
+            [SectionSerializer(section).data for section in self.sections]
+        ]

```

Comment:
This test seems useless because this is exactly the same way `_serialize_schedule` gets schedules, so explicitly declare the expected data here

Reply:
Yeah that's true, I just really didn't want to write it all out manually lol

Reply:
Actually I think I disagree. We shouldn't be testing the SectionSerializer's functionality (which we would be doing if we wrote it all the way out). What this is really testing is that it gets the corresponding sections id's, fetches the models, and serializes the _correct_ sections, not that it serializes the sections correctly. Hopefully this way if the `SectionSerializer` is broken, this will still pass, since the output will be the same for both (assuming it's still getting the same sections).

Reply:
That being said, for the `/schedules/generate` route, I should probably write out completely since I think the output in that case is relevant to the test (probably)